launch/launch/logging/__init__.py

```diff
@@ -30,123 +30,203 @@
+        """
+        Set up log handler factory.
+
+        :param new_log_handler_factory: a callable to build log file handler.
+           It takes a path to the log file and it must return a `logging.Handler`
+           variant with per logger formatting support.
+           See `launch.logging.handlers` module for further reference and easy reuse
+           of existing standard `logging.handlers` module handlers.
+           Defaults to regular log file handlers for logging if no factory is given.
+        """
+        self._log_handler_factory = new_log_handler_factory
+
+    def get_log_handler_factory(self):
+        """Get the log_handler_factory, generating it if necessary."""
+        if self._log_handler_factory is None:
+            if os.name != 'nt':
+                self._log_handler_factory = handlers.WatchedFileHandler
+            else:
+                self._log_handler_factory = handlers.FileHandler
+        return self._log_handler_factory
+
+    log_handler_factory = property(get_log_handler_factory, set_log_handler_factory)
+
+    def set_screen_format(self, screen_format, screen_style):
+        """
+        Set up screen formats.
+
+        For the ``screen_format`` argument there are a few aliases:
+
+          - 'default' to log verbosity level, logger name and logged message
+          - 'default_with_timestamp' to add timestamps to the 'default' format
+
+        :param screen_format: format specification used when logging to the screen,
+            as expected by the `logging.Formatter` constructor.
+            Alternatively, aliases for common formats are available, see above.
+        :param screen_style: the screen style used if no alias is used for
+            screen_format.
+            No style can be provided if a format alias is given.
+        """
+        if screen_format is not None:
+            if screen_format == 'default':
+                screen_format = '[{levelname}] [{name}]: {msg}'
+                if screen_style is not None:
+                    raise ValueError(
+                        'Cannot set a custom format style for the "default" screen format.'
+                    )
+            if screen_format == 'default_with_timestamp':
+                screen_format = '{created:.7f} [{levelname}] [{name}]: {msg}'
+                if screen_style is not None:
+                    raise ValueError(
+                        'Cannot set a custom format style for the '
+                        '"default_with_timestamp" screen format.'
+                    )
+            if screen_style is None:
+                screen_style = '{'
+            self.screen_formatter = logging.Formatter(
+                screen_format, style=screen_style
+            )
+            if self.screen_handler is not None:
+                self.screen_handler.setFormatter(self.screen_formatter)
+        else:
+            self.screen_formatter = None
+
+    def get_screen_handler(self):
```

Comment:
@clalancette I wonder if `logging` handlers' getters should module-level functions instead of part of the configuration. These _use_ the configuration but are not part of it IMO.

Reply:
The thing that breaks it is this: https://github.com/ros2/launch/pull/299/files#diff-01b1f102bf9f7e90ec6fd88bf1ac3780R178 .  There, we *do* hold onto the screen_handler so we can later set a screen_formatter on it.  At least as it stands, I think it needs to be part of the class.  Unless you have another idea here?

Reply:
Hmm that's true. Nevermind.